src/AbstractPluginManager.php

```diff
@@ -201,21 +139,143 @@ public function validate(mixed $instance)
     }
 
     /**
-     * Implemented for backwards compatibility only.
+     * {@inheritDoc}
+     */
+    public function has(string $id): bool
+    {
+        return $this->plugins->has($id);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function build(string $name, ?array $options = null): mixed
+    {
+        /** @psalm-suppress MixedAssignment Yes indeed, service managers can return mixed. */
+        $plugin = $this->plugins->build($name, $options);
+        $this->validate($plugin);
```

Comment:
Discussed in #182

I overall think we need to redesign this class, but I don't mind doing it in a patch addressing #182 directly.